crates/solver/src/solver.rs

```diff
@@ -567,12 +569,21 @@ pub async fn create(
                     slippage_calculator,
                 )))),
             };
+
+            let solver = match solver {
+                Some(solver) => solver,
+                None => {
+                    return None;
+                }
+            };
+
             shared(OptimizingSolver {
                 inner: solver,
                 post_processing_pipeline: post_processing_pipeline.clone(),
                 score_calculator,
             })
         })
+        .flatten()
```

Comment:
Instead of calling `.map().flatten()` `clippy` suggests to use `.filter_map()` instead.